lib/rules/jsx-curly-brace-presence.js

```diff
@@ -103,34 +127,42 @@ module.exports = {
         node: literalNode,
         message: 'Need to wrap this literal in a JSX expression.',
         fix: function(fixer) {
+          // Leave it to the author to fix as it can be fixed by either a
+          // real character or unicode
+          if (containsHTMLEntity(literalNode.raw)) {
+            return null;
+          }
+
           const expression = literalNode.parent.type === 'JSXAttribute' ?
-            `{"${escapeDoubleQuotes(
+            `{"${escapeDoubleQuotes(escapeBackslashes(
```

Comment:
We also need to escape line terminators here.

Reply:
I actually wanted to ask you about them. I know you mentioned them [here](https://github.com/yannickcr/eslint-plugin-react/issues/1449#issuecomment-332750344). 

> similarly for the line terminators: CR, LF, U+2028 and U+2029

I am not quite familiar with those. Do you type exactly those characters of any of the sets into the string to terminate a line? I tried having them in a JSX expression and it just displays those characters. Sorry about my lack of knowledge in the subject matter. 😀

Reply:
Press the Enter key to type a LF (`\n`) (or most likely CRLF `\r\n` on Windows).

You can also run one of the following in the Chrome or Firefox dev tools console to copy the character to the clipboard:

- `copy('\n')`
- `copy('\r')`
- `copy('\u2028')`
- `copy('\u2029')`

Then you can simply paste the character.

But since the tests are written as JavaScript strings you should be able to use JavaScript escapes:

```js
input: 'propWithLineTerminators="\n\r\u2028\u2029"',
```

Reply:
For the example you mentioned, the output would be `<div propWithLineTerminators="\\n\\r\\u2028\\u2029"/>`? Because these characters would be displayed as they are in a JSX attribute. 

In that case, wouldn't the parser have already handled that? 
![screen shot 2017-10-01 at 12 44 28 pm](https://user-images.githubusercontent.com/25375401/31058356-3af804be-a6a7-11e7-80e4-90b2b578264e.png)

Reply:
Actually, I think we should bail out if we find any line terminators at all, because of Babel's whitespace stripping thing.